catalog/core/catalog-core-subscriptionstore/src/main/java/org/codice/ddf/catalog/subscriptionstore/SubscriptionCache.java

```diff
@@ -0,0 +1,234 @@
+        notEmpty(messageBody, "messageBody cannot be null or empty");
+        notEmpty(callbackUrl, "callbackUrl cannot be null or empty");
+        notEmpty(subscriptionId, "subscriptionId cannot be null or empty");
+
+        if (!contains(subscriptionId, type)) {
+            LOGGER.debug("Target for subscription update [{}|{}] does not exist",
+                    subscriptionId,
+                    type);
+            return;
+        }
+
+        delete(subscriptionId, type);
+
+        SubscriptionMetadata metadata = new SubscriptionMetadata(type,
+                messageBody,
+                callbackUrl,
+                subscriptionId);
+
+        createSubscriptionLocally(metadata, subscription);
+        subscriptionStore.insert(metadata);
+    }
+
+    @Nullable
+    @Override
+    public Subscription delete(String subscriptionId, String type) {
+        notEmpty(subscriptionId, "subscriptionId cannot be null or empty");
+        notEmpty(type, "type cannot be null or empty");
+
+        if (!contains(subscriptionId, type)) {
+            return null;
+        }
+
+        SubscriptionMetadata metadata = deleteSubscriptionLocally(subscriptionId);
+        subscriptionStore.delete(subscriptionId);
+
+        return metadata.getSubscription()
+                .orElseThrow(() -> new SubscriptionStoreException(format(
+                        "Subscription [%s] was not cached",
+                        metadata.getId())));
+    }
+
+    @Override
+    public boolean contains(String subscriptionId, String type) {
+        return subscriptions.containsKey(subscriptionId) && type.equals(subscriptions.get(
+                subscriptionId)
+                .getType());
+    }
+
+    void createSubscriptionLocally(SubscriptionMetadata metadata) {
+        createSubscriptionLocally(metadata, reviveSubscription(metadata));
+    }
+
+    private void createSubscriptionLocally(SubscriptionMetadata metadata,
+            Subscription subscription) {
+        Dictionary<String, String> properties = new Hashtable<>();
+        properties.put(SUBSCRIPTION_ID, metadata.getId());
+        properties.put(EVENT_ENDPOINT,
+                metadata.getCallbackUrl()
+                        .toString());
+
+        ServiceRegistration registration =
+                getBundleContext().registerService(Subscription.class.getName(),
+                        subscription,
+                        properties);
+
+        if (registration != null) {
+            LOGGER.debug("Subscription [{}|{}|{}] registered with bundle ID = {} ",
+                    metadata.getId(),
+                    metadata.getType(),
+                    metadata.getCallbackUrl()
+                            .toString(),
+                    registration.getReference()
+                            .getBundle()
+                            .getBundleId());
+            metadata.setRegistration(registration);
+            subscriptions.put(metadata.getId(), metadata);
```

Comment:
Caught an omission: Below `metadata.setRegistration(registration)` there should be a call to `metadata.setSubscription(subscription)`.